src/optimizers/optimizer.cpp

```diff
@@ -76,9 +76,7 @@ optimizer& optimizer::operator=(const optimizer& other) {
   m_gradient_allreduce_needed = other.m_gradient_allreduce_needed;
   m_gradient_allreduce_started = other.m_gradient_allreduce_started;
   m_gradient_allreduce_finished = other.m_gradient_allreduce_finished;
-  #ifdef LBANN_NBALLREDUCE_GRADIENT
```

Comment:
We no longer check for the `LBANN_NBALLREDUCE_GRADIENT` macro. LBANN will always use non-blocking allreduces if they are available.